### DISCUSSION TITLE
Refresh Token again error 400

### DISCUSSION THREAD

**Discussion by MrCasquette** [directus/directus#16895] [Archive]
I use SDK on a nextjs app. Everything works fine until i refresh the page. Then when i want to logout I got this weird error that telling "refresh_token" is required in either the JSON payload or Cookie.... If i refresh the page again it's all gone... WTF 😅
I tried everything. Json mode, cookie mode, secured cookie, etc etc...
Please I really need a pretty auth on me websites (all my next project does this)

**Comment by rijkvanzanten** [comment]
> Then when i want to logout I got this weird error that telling "refresh_token" is required in either the JSON payload or Cookie....

Sounds like the `/auth/refresh` endpoint was called with no token. In this case that makes sense as you've logged out, so the auto-refresh on load kicks in and tries to refresh the session. The 400 in this case is expected 👍🏻
